production/helm/values.yaml

```diff
@@ -132,18 +139,22 @@ promtail:
     tag: latest
     pullPolicy: Always # Always pull while in BETA
 
+  port: 3101
+
+  # Extra volumes to scrape logs from
   volumes:
-    - name: varlog
+    - name: pods
       hostPath:
-        path: /var/log
-    - name: varlibdockercontainers
+        path: /var/log/pods
+    - name: docker
       hostPath:
         path: /var/lib/docker/containers
 
   volumeMounts:
-    - name: varlog
-      mountPath: /var/log
-    - name: varlibdockercontainers
+    - name: pods
+      mountPath: /var/log/pods
```

Comment:
This seems like a specialisation. Are we constraining this default mount too much?

Reply:
More tightened would be good from a security stand point. I don't know if all runtimes store the actual logs in /var/log/pods though.

I can confirm the log files are actually stored in /var/log/pods on containerd 1.2.4 with k8s 1.13.3.

The logs on docker 18.06.2-ce with k8s 1.13.3 are all symlinks in /var/log/pods to /var/log/docker/containers/. Verified on a recent minikube.

So, there does seem to be a good benefit security wise to use containerd over dockerd for this. Docker mixes logs along with other parts of the container. containerd lets you send only the logs to the log shipper, not the whole container.

Reply:
@davkal My reasoning is:
1. The mounted volumes should match the scrape config and right now the scrape config only uses /var/log/pods. If later the scrape config is changed, we can adjust it.
2. User can always add/remove/change volumes via `promtail.volumes`
3. Security as @kfox1111 mentions

Reply:
SGTM